pkg/apis/core/validation/validation.go

```diff
@@ -2507,16 +2507,70 @@ func validateDNSPolicy(dnsPolicy *core.DNSPolicy, fldPath *field.Path) field.Err
 	allErrors := field.ErrorList{}
 	switch *dnsPolicy {
 	case core.DNSClusterFirstWithHostNet, core.DNSClusterFirst, core.DNSDefault:
-		break
+	case core.DNSNone:
+		if !utilfeature.DefaultFeatureGate.Enabled(features.CustomPodDNS) {
+			allErrors = append(allErrors, field.Invalid(fldPath, dnsPolicy, "DNSPolicy: can not use 'None', custom pod DNS is disabled by feature gate"))
+		}
 	case "":
 		allErrors = append(allErrors, field.Required(fldPath, ""))
 	default:
 		validValues := []string{string(core.DNSClusterFirstWithHostNet), string(core.DNSClusterFirst), string(core.DNSDefault)}
+		if utilfeature.DefaultFeatureGate.Enabled(features.CustomPodDNS) {
+			validValues = append(validValues, string(core.DNSNone))
+		}
 		allErrors = append(allErrors, field.NotSupported(fldPath, dnsPolicy, validValues))
 	}
 	return allErrors
 }
 
+const (
+	// Below constants define the limits for PodDNSConfig.
```

Comment:
// Limits on various DNS parameters. These are derived from restrictions in Linux libc name resolution handling.

// Maximum number of DNS nameservers.

etc.

avoid abbreviations

Reply:
Thanks, done.